apps/angular/bug-effect-signal/src/app/app.component.ts

```diff
@@ -28,7 +28,24 @@ export class AppComponent {
 
   constructor() {
     effect(() => {
-      if (this.name() || this.age() || this.address()) {
+      /* 
+        This effect executes only once when name is true, because the value of name doesn't change.
+        Effects run when a signal inside changes, since name value is the same as before, the effect doesn't trigger.
+        If we click on age first, the effect will run again after click on name. 
+        Because there is no signal value which has the same value as before.
+      */
+      // if (this.name() || this.age() || this.address()) {
+      //   alert('Checkbox was checked');
+      // }
+      /* 
+        This effect should run if any of these 3 values is true, you have to define those values.
+        With this all signal values are interpreted as relevant values for this effect.
+        So if any of these values changes, the effect will run.
+      */
+      const name = this.name();
```

Comment:
Yes but i wasn't really sure what is part of the challenge and what's not. This is the simplest solution without changing the code a lot. As thomas already mentioned, this could be solved with a computed signal as well. I already have some things in mind, i'll update the docs for this challenge.

Reply:
I think we need clarification on the requirements.

Reply:
@LMFinney you are right, but I didn't go that far 😅
My goal was to trigger the effect when we check or uncheck one checkbox which is not the case if the first one is true. But I didn't think of your case @LMFinney; So I should change the instruction to be clearer indeed 👍